gpflow/kernels/mo_kernels.py

```diff
@@ -80,6 +86,10 @@ def __init__(self, kernel: Kernel, output_dimensionality: int):
         self.kernel = kernel
         self.P = output_dimensionality
 
+    @property
+    def num_output_dims(self):
+        return self.P
```

Comment:
Could we unify this to all be one thing? `self.output_dim = output_dim` in the `__init__`, then no further property is needed?:)

Reply:
sure, it means creating an __init__ in some cases where there isn't one so far but yes. I like the abstract property because it means new subclasses can't forget to define the property.

Reply:
Mainly I want to avoid property, attribute, and `__init__` argument all being different names...

Reply:
what do you mean? Also what do you think of `num_output_dims`? I think its a bit long, but ive met others who dont like `output_dims` (which generally i prefer)

Reply:
My first choice would be `output_dim` (no plural, it's just a scalar). `num_output_dim(s)` is ok too.

Reply:
We could follow a standard convention, instead of making our own:

https://www.tensorflow.org/api_docs/python/tf/keras/backend/ndim
https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.ndim.html

Reply:
I don't think that would be sufficiently unambiguous, given that there's input_dim as well (even if that one can now mostly be implicit)

Reply:
confusingly both of those actually use `ndim`  in a differnet way to us using `output_dim`.  in both cases `ndim` is actually the *rank* of the tensor, where as we want the *dimensionality* of a rank-1 tensor (vector), which normally corresponds to one element of `.shape`. I think the current version is unambiguous.